.cirrus.yml

```diff
@@ -2,27 +2,121 @@ env:
+
+  GIT_SUB_MODULE: ""
+  GRADLE_USER_HOME: ${CIRRUS_WORKING_DIR}/.gradle
+
+linux_image_template: &LINUX_IMAGE
+  image: ${CIRRUS_AWS_ACCOUNT}.dkr.ecr.eu-central-1.amazonaws.com/base:j17-g7-latest
+  cluster_name: ${CIRRUS_CLUSTER_NAME}
+  region: eu-central-1
+  namespace: default
+  use_in_memory_disk: true
+
+linux_2_cpu_4G_template: &LINUX_2_CPU_4G
+  eks_container:
+    <<: *LINUX_IMAGE
+    cpu: 2
+    memory: 4G
+
+linux_4_cpu_6G_java_17_template: &LINUX_4_CPU_6G_JAVA_17
+  eks_container:
+    <<: *LINUX_IMAGE
+    image: ${CIRRUS_AWS_ACCOUNT}.dkr.ecr.eu-central-1.amazonaws.com/base:j17-g7-latest
+    cpu: 4
+    memory: 6G
+
 eks_container: &CONTAINER_DEFINITION
-  image: ${CIRRUS_AWS_ACCOUNT}.dkr.ecr.eu-central-1.amazonaws.com/base:latest
+  image: ${CIRRUS_AWS_ACCOUNT}.dkr.ecr.eu-central-1.amazonaws.com/base:j17-g7-latest
   cluster_name: ${CIRRUS_CLUSTER_NAME}
   region: eu-central-1
   namespace: default
 
+## Build tasks
+
+setup_gradle_cache_template: &SETUP_GRADLE_CACHE
+  gradle_cache:
+    folder: .gradle/caches
+  create_gradle_directory_script:
+    - mkdir -p "${CIRRUS_WORKING_DIR}/.gradle"
+
+cleanup_gradle_cache_script_template: &CLEANUP_GRADLE_CACHE_SCRIPT
+  cleanup_gradle_script:
+    - /usr/bin/find "${CIRRUS_WORKING_DIR}/.gradle/caches/" -name "*.lock" -type f -delete
+    - rm -rf "${CIRRUS_WORKING_DIR}/.gradle/caches/4.10.2/"
+    - rm -rf "${CIRRUS_WORKING_DIR}/.gradle/caches/journal-1/"
+    - rm -rf "${CIRRUS_WORKING_DIR}/.gradle/caches/build-cache-1/"
+
 build_task:
-  eks_container:
-    <<: *CONTAINER_DEFINITION
-    cpu: 1
-    memory: 1G
+  <<: *LINUX_2_CPU_4G
+  <<: *SETUP_GRADLE_CACHE
   env:
-    # Possible values for ARTIFACTORY_DEPLOY_REPO: sonarsource-private-qa, sonarsource-public-qa
-    ARTIFACTORY_DEPLOY_REPO: FIXME
+    ARTIFACTORY_DEPLOY_REPO: sonarsource-public-qa
     ARTIFACTORY_DEPLOY_USERNAME: VAULT[development/artifactory/token/${CIRRUS_REPO_OWNER}-${CIRRUS_REPO_NAME}-qa-deployer username]
     ARTIFACTORY_DEPLOY_PASSWORD: VAULT[development/artifactory/token/${CIRRUS_REPO_OWNER}-${CIRRUS_REPO_NAME}-qa-deployer access_token]
     DEPLOY_PULL_REQUEST: "true"
   build_script:
     - source cirrus-env BUILD
-    - regular_build_...
+    # todo fix by using the script to build analyze and deploy
+    - ./gradlew build
+  <<: *CLEANUP_GRADLE_CACHE_SCRIPT
+
+
+### QA tasks
+
+only_if_sonarsource_qa_template: &ONLY_IF_SONARSOURCE_QA
+  only_if: $CIRRUS_USER_COLLABORATOR == 'true' && $CIRRUS_TAG == "" && ($CIRRUS_PR != "" || $CIRRUS_BRANCH == "master" || $CIRRUS_BRANCH =~ "branch-.*" || $CIRRUS_BRANCH =~ "dogfood-on-.*")
+
+qa_task_filter_template: &QA_TASK_FILTER
+  depends_on:
+    - build
+  <<: *ONLY_IF_SONARSOURCE_QA
+
+gradle_its_template: &GRADLE_ITS_TEMPLATE
+  <<: *SETUP_GRADLE_CACHE
+  run_its_script:
+    - |
+      if [ -n "${GIT_SUB_MODULE}" ]; then        
+        git submodule update --init --depth 1 "${GIT_SUB_MODULE}"
+      fi
```

Comment:
Does this need to be conditional? Can we just update the submodules whatever happens?

Reply:
Apparently, it is required by the ruling tests only; it can be added to that task explicitly